File: examples/transpiler/ejemplo.py
from abc import ABC, abstractmethod

class Ser(ABC):
    @abstractmethod
    def actuar(self):
        pass

class Humano(Ser):
    def __init__(self, nombre):
        self.nombre = nombre
    def actuar(self):
        print(f"{self.nombre} piensa y siente.")

class Robot(Ser):
    def __init__(self, id):
        self.id = id
    def actuar(self):
        print(f"Robot {self.id} ejecuta instrucciones.")

def main():
    seres = [Humano("Ari"), Robot("R2")]
    for s in seres:  # Duck typing / polimorfismo: basta con que implementen actuar()
        s.actuar()

if __name__ == "__main__":
    main()

File: examples/transpiler/ejemplo.rs
trait Ser {
    fn actuar(&self);
}

struct Humano {
    nombre: String,
}

impl Ser for Humano {
    fn actuar(&self) {
        println!("{} piensa y siente.", self.nombre);
    }
}

struct Robot {
    id: String,
}

impl Ser for Robot {
    fn actuar(&self) {
        println!("Robot {} ejecuta instrucciones.", self.id);
    }
}

fn main() {
    let seres: Vec<Box<dyn Ser>> = vec![
        Box::new(Humano { nombre: "Ari".into() }),
        Box::new(Robot { id: "R2".into() }),
    ];

    for s in seres.iter() {
        s.actuar(); // dispatch dinámico vía trait object
    }
}
